### ISSUE TITLE
How to validate sub document field?

### ISSUE THREAD

**Issue by sagarguhe** [sokil/php-mongo#139]
How do I validate sub document field? for example if I have document like this:

`{`
  `"name": "John",`
  `"email": "john.doe@example.com",`
  `"addresses": [`
    `{`
      `"street": "field required",`
      `"locality": "",`
      `"city": "",`
      `"state": "",`
      `"country": ""`
    `},`
    `{`
      `"street": "field required",`
      `"locality": "",`
      `"city": "",`
      `"state": "",`
      `"country": ""`
    `}`
  `],`
  `"profile": {`
    `"profile_picture": "myavtar.jpg",`
    `"skills": "php,mongo,mysql",`
    `"years_exp": "field required"`
 ` }`
`}`

In the above document I need `street` from addresses and `years_exp` from profile must be filled by the user.

So how do i validate these fields?

**Comment by sokil** [comment]
You need to implement `Structure` with validation rules, and validate this instance. Currently, embedded documents validates only on Document::set() or manually before adding in any way to document

**Comment by sagarguhe** [comment]
I found a way to validate a `profile.years_exp` (<--- so you have to write exactly like that) but not able to validate `addresses.n.street` where `n` stands for `index` of an element.

**Comment by sokil** [comment]
yes. if emdemmed document can be accessed directly, validation may be configured in document. for validation of liost of embedded documents you need to have ebbedded document class, extended from `Structure`. Currently this is the only way.
